server/src/main/java/io/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -253,6 +255,26 @@ public void start() throws Exception
           @Override
           public void stop()
           {
+            if (activeConnections.get() > 0) {
+              long gracefulShutDownTime = (long) server.getAttribute(GRACEFUL_SHUTDOWN_TIME);
+              log.info("Waiting for [%s] milliseconds for active requests to be zero, current active requests=[%s]",
+                  gracefulShutDownTime, activeConnections.get());
+              try {
+                Thread.sleep(gracefulShutDownTime);
```

Comment:
A couple questions.

1. Will the server always unannounce itself before entering this `stop()` code? How is that assured? If it happens, that's good; but if not, that would mean new requests could keep coming in continuously even while waiting for shutdown. Inevitably the last few will fail.
2. Could this code check `activeConnections` every few ms, rather than always sleeping for the gracefulShutdownTimeout? Otherwise, it looks like an extra 5s of potentially needless delay.
3. If both of the above are assured, then I think it could make sense to set the default gracefulShutdownTimeout to somewhat longer, like a few minutes.